### ISSUE TITLE
GWEN is failing with error when we use "-Drp.rerun=true" for report portal

### ISSUE THREAD

**Issue by sse2ebdd** [gwen-interpreter/gwen-web#92]
**Steps to reproduce -**

1. Create simple feature file with below content
```
Feature: Test

Scenario: Scenario name
Given something is "value"
And something1 is "value2"
And something2 is "value3"
```
2. When we run GWEN using -Drp.rerun=false, feature file passes
3. But when we run GWEN using -Drp.rerun=true, it fails with below error
```
Closing Report Portal connection..ERROR - Start test item completed with error
java.lang.IllegalStateException: Executor pool shut down
        at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint.executeInternal(HttpClientRestEndpoint.java:491)
        at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint.executeRequest(HttpClientRestEndpoint.java:411)
        at com.epam.reportportal.restendpoint.http.proxy.RestEndpointInvocationHandler.executeRestMethod(RestEndpointInvocationHandler.java:59)
        at com.epam.reportportal.restendpoint.http.proxy.RestEndpointInvocationHandler.invoke(RestEndpointInvocationHandler.java:43)
        at com.sun.proxy.$Proxy6.startTestItem(Unknown Source)
        at com.epam.reportportal.service.LaunchImpl.lambda$startTestItem$8(LaunchImpl.java:268)
        at io.reactivex.internal.operators.maybe.MaybeFlatten$FlatMapMaybeObserver.onSuccess(MaybeFlatten.java:88)
        at io.reactivex.internal.operators.maybe.MaybeCache.onSuccess(MaybeCache.java:92)
        at io.reactivex.internal.operators.maybe.MaybeCreate$Emitter.onSuccess(MaybeCreate.java:73)
        at com.epam.reportportal.service.LaunchImpl.lambda$null$4(LaunchImpl.java:136)
        at io.reactivex.internal.operators.maybe.MaybeCallbackObserver.onSuccess(MaybeCallbackObserver.java:71)
        at io.reactivex.internal.operators.maybe.MaybeCache.onSuccess(MaybeCache.java:92)
        at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeOnMaybeObserver.onSuccess(MaybeSubscribeOn.java:91)
        at io.reactivex.internal.operators.maybe.MaybePeek$MaybePeekObserver.onSuccess(MaybePeek.java:122)
        at io.reactivex.internal.operators.maybe.MaybePeek$MaybePeekObserver.onSuccess(MaybePeek.java:122)
        at io.reactivex.internal.operators.flowable.FlowableSingleMaybe$SingleElementSubscriber.onComplete(FlowableSingleMaybe.java:104)
        at io.reactivex.internal.operators.flowable.FlowableRetryPredicate$RetrySubscriber.onComplete(FlowableRetryPredicate.java:104)
        at io.reactivex.internal.subscriptions.DeferredScalarSubscription.complete(DeferredScalarSubscription.java:134)
        at io.reactivex.internal.operators.maybe.MaybeToFlowable$MaybeToFlowableSubscriber.onSuccess(MaybeToFlowable.java:70)
        at io.reactivex.internal.operators.maybe.MaybeFlatten$FlatMapMaybeObserver$InnerObserver.onSuccess(MaybeFlatten.java:119)
        at io.reactivex.internal.operators.maybe.MaybeJust.subscribeActual(MaybeJust.java:36)
        at io.reactivex.Maybe.subscribe(Maybe.java:4290)
        at io.reactivex.internal.operators.maybe.MaybeFlatten$FlatMapMaybeObserver.onSuccess(MaybeFlatten.java:96)
        at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeOnMaybeObserver.onSuccess(MaybeSubscribeOn.java:91)
        at io.reactivex.internal.operators.maybe.MaybeCreate$Emitter.onSuccess(MaybeCreate.java:73)
        at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint$1.subscribe(HttpClientRestEndpoint.java:536)
        at io.reactivex.internal.operators.maybe.MaybeCreate.subscribeActual(MaybeCreate.java:45)
        at io.reactivex.Maybe.subscribe(Maybe.java:4290)
        at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54)
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38)
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
  [1s 068ms] Closed

INFO - Report Portal launch report generated: http://IP:PORT/ui/#PROJECT/launches/all/LAUNCH_ID
ERROR - Strict behavior rules violation: Given-When-Then order not satisfied by steps in Scenario [at BDD/features/test.feature:22]
```
4. Also GWEN is giving "Strict behavior rules violation" error, even if we have defined "gwen.behavior.rules=lenient" in properties.

Env
Gwen Version - 3.0.9
Chrome Driver - Latest (96)

**Comment by bjuric** [comment]
I was not able to replicate this. Are there any other errors appearing in the log prior to this error? It looks like no properties are being loaded at all in your case.

**Comment by sse2ebdd** [comment]
Error mentioned in above comment is the only error I get on running GWEN.
For replicating this issue, make sure you give "rp" in reports parameter and use "-Drp.rerun=true".

Command which I used --> 
`gwen.bat -r reports -f "rp,html" -c ".\BDD\properties\gwen.properties" -Drp.rerun=true -Drp.uuid=USER_UUID -Drp.endpoint=http://RP_HOSTNAME:PORT BDD\features\test.feature`

**Comment by bjuric** [comment]
I was able to successfully use `-Drp.rerun-true` and got a rerun in report portal as expected. So I cannot replicate this.

**Comment by sse2ebdd** [comment]
I tried running GWEN multiple times using command mentioned in above comment. (-Drp.rerun=true)
Sometimes I get this error --> "java.lang.IllegalStateException: Executor pool shut down"
And sometimes I only get error --> "ERROR - Strict behavior rules violation: Given-When-Then order not satisfied by steps in Scenario [at BDD/features/test.feature:22]"

So this error "Strict behavior rules violation:" implies that properties are not picked up by GWEN.
But if that's the case, then same feature file should not pass when I use -Drp.rerun=false.
I tried running same feature file multiple times using -Drp.rerun=false, and it passes every time.

Also, if I run feature without "rp" in report format, feature passes every time.

**Comment by sse2ebdd** [comment]
I think I got the issue.
When we mention "rp" in reports format and -Drp.rerun=true, GWEN tries to pick **gwen.properties** from current path from which you have started GWEN. Which means it does not respect properties mentioned in -c/-p parameter.

**Exact Steps to replicate -**
1. Create feature with below content on path --> C:\gwen-web-3.0.9\bin\test.feature
```
Feature: Test

Scenario: Scenario name
Given something is "value"
When something1 is "value2"
Then something2 is "value3"
```
2. Create 2 properties files with below content on two different paths **"C:\gwen-web-3.0.9\bin\gwen.properties"** AND **"C:\gwen-web-3.0.9\bin\properties\gwen.properties"**
```
gwen.behavior.rules=lenient
gwen.feature.mode=imperative
gwen.target.browser=chrome

rp.launch = TEST_LAUNCH
rp.project = TEST_PROJECT
```
3. Now, cd C:\gwen-web-3.0.9\bin\ and run below command
`gwen -r reports -f "rp,html" -c "properties\gwen.properties" -Drp.rerun=true -Drp.uuid=RP_USER_UUID -Drp.endpoint=http://HOSTNAME:PORT test.feature`

Feature file will get passed **because we have gwen.properties in current path**, which is C:\gwen-web-3.0.9\bin\

4. Now rename properties file in current path to gwen1.properties. And again run same command.
You can now see GWEN gives error --> Declarative feature mode violation: DSL step not permitted in feature [at BDD/features/test.feature:4]
Because GWEN didn't find any properties in current path and it doesn't respect properties which we mentioned -c "properties\gwen.properties".

I hope this helps to replicate issue.

**Comment by bjuric** [comment]
Thanks for the clarifying the issue. Fixed in v[3.0.10](https://github.com/gwen-interpreter/gwen-web/releases/tag/v3.0.10)

**Comment by sse2ebdd** [comment]
Verified this issue on GWEN v3.0.10.
It's working as expected, hence closing this issue.
